docs/antrea-network-policy.md

```diff
@@ -1037,13 +887,82 @@ spec:
         name: AllowToCoreDNS
 ```
 
-The above example allows all Pods from Namespace "default" to connect to all "core-dns"
+The above example allows all Pods from Namespace "default" to connect to all "kube-dns"
 Pods from Namespace "kube-system" on TCP port 53.
 
-## FQDN based filtering
+### Matching self Namespace for appliedTos
 
-Antrea-native policy accepts a `fqdn` field to select Fully Qualified Domain Names (FQDNs),
-specified either by exact name or wildcard expressions, when defining `egress` rules.
+The `namespaces` field allows users to perform advanced matching on Namespace objects that cannot 
+be done via label selectors. Currently, the `namespaces` field has only one matching strategy, `Self`. 
+If set to `Self`, for each Pod affected by `appliedTo` of the policy/rule, this field selects the same
+Namespace that the Pod is in. It enables policy writers to create per-Namespace rules within a
+single policy.
```

Comment:
I feel like the previous version of this paragraph had a bit more details?

Reply:
Yes there was a more verbose sentence, which I find hard to understand when I read it now (even though I wrote it myself). That's why I put two addition sample specs, one with just the `namespaces` field and the other being the `namespaces`+`podSelector` to illustrate how it works. I was hoping that it will be more readable.